### DISCUSSION TITLE
`extend.spacing` applied inconsistently to sub-styles

### DISCUSSION THREAD

**Discussion by haydenlinder** [tailwindlabs/tailwindcss#7600] [Help]
**What version of Tailwind CSS are you using?**

Tailwind playground

**What build tool (or framework if it abstracts the build tool) are you using?**

Tailwind playground

**What version of Node.js are you using?**

Tailwind playground

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/VR5Qrsa4As

**Describe your issue**

After configuring `custom` spacing in tailwind config, I expect that `min-h-custom`, `min-w-custom`, etc should behave the same, but that is not the case.

**Comment by simonswiss** [comment]
Hey!

We don't do this because the `spacing` scale has far too many values for something like `min-width` or `min-height` utilities.

You can see in the default config that we _do_ apply the `spacing` scale to `max-height/width`:

https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/defaultConfig.stub.js#L612-L613

However, `min-height/width` in the theme only have a handful of options:

https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/defaultConfig.stub.js#L641-L655

Again that's because we don't think that for most projects you'd ever need the entire spacing scale to choose from for `min-height/width`.

## Extending the `min-height/width` with the spacing scale

If you however do want to have access to all of these, in your config file you can extend your `minHeight` or `minWidth` theme objects with the spacing scale like so:

```js
// tailwind.config.js
module.exports = {
  theme: {
    extend: {
      minHeight: ({theme}) => ({
        ...theme('spacing')
      })
    }
  }
}
```

By defining the `minHeight` options as a function, you can access the `theme` function which lets you access other properties of the theme.

So here, I spread the entire `spacing` scale into the `minHeight` options with `...theme('spacing')`.

This means that now I have access to all the spacing values as `min-h-{x}` utility classes:

![image](https://user-images.githubusercontent.com/485747/155245137-23503f62-874b-4902-98ce-e4fa9f23c007.png)

If you've added some custom values to your `spacing` scale in the theme, these custom values will also be available in in your min-height utilities:

![image](https://user-images.githubusercontent.com/485747/155245392-0d0041ca-b11e-4521-af2f-a978a320eb2a.png)


![image](https://user-images.githubusercontent.com/485747/155245371-a8ca6e96-a25f-4086-81b4-0cdf45ec2018.png)

Hope it helps! Here's the Tailwind Play I took the screenshots from, as a reference:

https://play.tailwindcss.com/66UhsNeQ0M?file=config

Moving this to our GitHub discussions since I think this is an interesting topic that may help folks having similar expectations/assumptions.

Have a great day! 🎉

**Comment by code23-isaac** [comment]
Not sure why having the options available is an issue with JIT as core? Seems an overly opinionated call to break the convention on these utilities, but glad we can configure it so easily

**Reply by adamwathan** [reply]
Just haven't revisited the defaults since JIT was introduced. Definitely something we intend to do at some point, probably for v4 👍  Agree we can safely beef these up a lot now.

**Comment by MisterJimson** [comment]
Fixed in v3.4!

https://tailwindcss.com/blog/tailwindcss-v3-4#extended-min-width-max-width-and-min-height-scales
